src/daemon/mod.rs

```diff
@@ -141,82 +144,200 @@ pub async fn start_interruptable(opts: CliOpts, config: Config) -> anyhow::Resul
 // Garbage collection interval, currently set at 10 hours.
 const GC_INTERVAL: Duration = Duration::from_secs(60 * 60 * 10);
 
-/// Starts daemon process
-pub(super) async fn start(
-    opts: CliOpts,
-    config: Config,
-    shutdown_send: mpsc::Sender<()>,
-) -> anyhow::Result<()> {
+type DbType = ManyCar<Arc<ParityDb>>;
+
+struct DbMetadata {
+    db_root_dir: PathBuf,
+    forest_car_db_dir: PathBuf,
+}
+
+fn do_startup_miscs_and_get_timestamp(
+    opts: &CliOpts,
+    config: &Config,
+) -> anyhow::Result<chrono::DateTime<chrono::Utc>> {
+    let start_time = chrono::Utc::now();
```

Comment:
I'd omit this from the function body and move it up (to `start`). The method name would be more coherent.